windows_service/datadog_checks/windows_service/data/conf.yaml.example

```diff
@@ -13,17 +13,29 @@ init_config:
 #
 instances:
 
-    ## @param services - list of strings - required
+    ## @param services - (list of string or mapping) - required
     ## List of services to monitor e.g. Dnscache, wmiApSrv, etc.
     ##
-    ## If any service is set to `ALL`, all services registered with the SCM will be monitored.
+    ## Services can also be selectively monitored by their startup type.
+    ## Possible values for the `startup_type` pattern:
+    ##   - disabled
+    ##   - manual
+    ##   - automatic
+    ##   - automatic_delayed_start
+    ##
+    ## If any service is set to `ALL`, all services registered with the SCM will be monitored, and
+    ## all other patterns provided in this instance will be ignored.
+    ##
+    ## Patterns are case insensitive. `ALL` is case sensitive.
     ##
-    ## This matches all services starting with service, as if service.* is configured.
+    ## NOTE: The configured patterns will match all services starting with the pattern. For example,
+    ## if `service` is provided, it will match all services starting with `service` as if service.* is configured.
     ## For an exact match, use ^service$
     #
   - services:
       - <SERVICE_NAME_1>
       - <SERVICE_NAME_2>
+      - startup_type: ^automatic$
```

Comment:
`- startup_type: ^automatic$`
Interesting aspect. The number of states is small. Do we really need regex?

Reply:
probably not, I kept it for consistency.
Though I don't like how this config will match both `automatic` and `automatic_delayed_start`.
```
services:
  - startup_type: automatic
```

I'd worry about breaking some customer's configs if we changed the match behavior now though. That's why in the yaml examples I was explicit about  `^automatic$`.

Reply:
I changed my mind.

I decided that between the following two cases, I'd rather have the second.

case 1 (regex match): 
customer who does not want to see delayed start services to change their config from
```
services:
  - startup_type: automatic
```
to
```
services:
  - startup_type: ^automatic$
```

case 2 (string match):
customer who wants to monitor both automatic and delayed start services and has this config
```
services:
  - startup_type: automatic
```
to change their config to
```
services:
  - startup_type: automatic
  - startup_type: automatic_delayed_start
```